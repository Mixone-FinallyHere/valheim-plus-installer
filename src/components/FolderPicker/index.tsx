import Fade from "@material-ui/core/Fade";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import { makeStyles } from "@material-ui/core/styles";
import Tooltip from "@material-ui/core/Tooltip";
import FolderIcon from "@material-ui/icons/Folder";
import PropTypes from "prop-types";
import React from "react";

interface IFolderPickerProps {
  folder: string;
  handleClick: (event: React.MouseEvent | null) => void;
  label: string;
  toolTip: string;
}

const useStyles = makeStyles(theme => ({
  listItem: {
    backgroundColor: theme.palette.background.paper,
    cursor: "pointer",
    margin: "5px 0px",
    borderRadius: 5
  }
}));

function FolderPicker(props: IFolderPickerProps): React.ReactElement {
  const classes = useStyles();

  return (
    <Tooltip
      title={props.toolTip}
      color="secondary"
      placement="top-start"
      enterDelay={500}
      leaveDelay={200}
      TransitionComponent={Fade}
      TransitionProps={{ timeout: 600 }}
    >
      <ListItem className={classes.listItem} onClick={props.handleClick}>
        <ListItemIcon>
          <FolderIcon />
        </ListItemIcon>
        <ListItemText primary={props.label} secondary={props.folder} />
      </ListItem>
    </Tooltip>
  );
}

FolderPicker.propTypes = {
  folder: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  toolTip: PropTypes.string.isRequired
};

export default FolderPicker;
