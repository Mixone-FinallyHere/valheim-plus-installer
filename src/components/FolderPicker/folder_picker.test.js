import React from "react";
import FolderPicker from ".";
import { render } from "@testing-library/react";

it("renders without crashing", () => {
  const { asFragment } = render(
    <FolderPicker folder={"/foo/bar"} handleClick={jest.fn()} label={"foo"} toolTip={"bar"} />
  );

  expect(asFragment()).toMatchSnapshot();
});
