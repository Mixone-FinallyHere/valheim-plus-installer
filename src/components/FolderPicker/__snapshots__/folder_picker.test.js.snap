// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <li
    class="MuiListItem-root makeStyles-listItem-1 MuiListItem-gutters"
    color="secondary"
    title="bar"
  >
    <div
      class="MuiListItemIcon-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
        />
      </svg>
    </div>
    <div
      class="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        foo
      </span>
      <p
        class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        /foo/bar
      </p>
    </div>
  </li>
</DocumentFragment>
`;
